lxdock/container.py

```diff
@@ -47,8 +69,12 @@ class Container:
     def __init__(self, project_name, homedir, client, **options):
         self.project_name = project_name
         self.homedir = homedir
-        self.client = client
         self.options = options
+        self.lxd_host = os.path.basename(get_container_host(self.lxd_name)).split(':')[0]
+        try:
+            self.client = get_client([get_container_host(self.lxd_name)])
```

Comment:
Since it's possible for the passed in `client` argument to be ignored, it might be better to make it optional and only take it if it's passed in explicitly.

```
def __init__(self, project_name, homedir, client=None, **options):
    ....
```